nomad/structs/workload_id_test.go

```diff
@@ -200,7 +201,7 @@ func TestNewIdentityClaims(t *testing.T) {
 		},
 		"job/group/task/alt-identity": {
 			Namespace: "default",
-			JobID:     "job",
+			JobID:     "parentJob",
 			TaskName:  "task",
 			Claims: jwt.Claims{
 				Subject:  "global:default:job:group:task:alt-identity",
```

Comment:
@schmichael I'm just now noticed we don't use the parent ID for the `Subject` claim... is that intention or did we repeat the same bug there?

Reply:
We repeated the same bug. I'm sure I just forgot about the distinction between Parent/Child IDs when writing it initially. :(

Reply:
I'll fix that in a separate PR.

Reply:
https://github.com/hashicorp/nomad/pull/23902